exercises/practice/palindrome-products/tests/palindrome-products.rs

```diff
@@ -40,84 +19,104 @@ fn palindrome_new_return_none() {
-fn find_smallest_palindrome_from_triple_digit_factors() {
-    process_smallest_case((100, 999), Some(10201));
+fn find_the_smallest_palindrome_from_triple_digit_factors() {
+    let output = palindrome_products(100, 999).map(|(min, _)| min.into_inner());
+    let expected = Some(10201);
+    assert_eq!(output, expected);
 }
 
 #[test]
 #[ignore]
-/// find the largest palindrome from triple digit factors
 fn find_the_largest_palindrome_from_triple_digit_factors() {
-    process_largest_case((100, 999), Some(906609));
+    let output = palindrome_products(100, 999).map(|(_, max)| max.into_inner());
+    let expected = Some(906609);
+    assert_eq!(output, expected);
 }
 
 #[test]
 #[ignore]
-/// find smallest palindrome from four digit factors
-fn find_smallest_palindrome_from_four_digit_factors() {
-    process_smallest_case((1000, 9999), Some(1002001));
+fn find_the_smallest_palindrome_from_four_digit_factors() {
+    let output = palindrome_products(1000, 9999).map(|(min, _)| min.into_inner());
+    let expected = Some(1002001);
+    assert_eq!(output, expected);
 }
 
 #[test]
 #[ignore]
-/// find the largest palindrome from four digit factors
 fn find_the_largest_palindrome_from_four_digit_factors() {
-    process_largest_case((1000, 9999), Some(99000099));
+    let output = palindrome_products(1000, 9999).map(|(_, max)| max.into_inner());
+    let expected = Some(99000099);
+    assert_eq!(output, expected);
 }
 
 #[test]
 #[ignore]
-/// empty result for smallest if no palindrome in the range
 fn empty_result_for_smallest_if_no_palindrome_in_the_range() {
-    process_smallest_case((1002, 1003), None);
+    let output = palindrome_products(1002, 1003).map(|(min, _)| min.into_inner());
+    let expected = None;
+    assert_eq!(output, expected);
 }
 
 #[test]
 #[ignore]
-/// empty result for largest if no palindrome in the range
 fn empty_result_for_largest_if_no_palindrome_in_the_range() {
-    process_largest_case((15, 15), None);
+    let output = palindrome_products(15, 15).map(|(_, max)| max.into_inner());
+    let expected = None;
+    assert_eq!(output, expected);
 }
 
 #[test]
 #[ignore]
-/// error result for smallest if min is more than max
 fn error_result_for_smallest_if_min_is_more_than_max() {
-    process_smallest_case((10000, 1), None);
+    let output = palindrome_products(10000, 1).map(|(min, _)| min.into_inner());
+    let expected = None;
+    assert_eq!(output, expected);
 }
 
 #[test]
 #[ignore]
-/// error result for largest if min is more than max
 fn error_result_for_largest_if_min_is_more_than_max() {
-    process_largest_case((2, 1), None);
+    let output = palindrome_products(2, 1).map(|(_, max)| max.into_inner());
+    let expected = None;
+    assert_eq!(output, expected);
+}
+
+#[test]
+#[ignore]
+fn smallest_product_does_not_use_the_smallest_factor() {
+    let output = palindrome_products(3215, 4000).map(|(min, _)| min.into_inner());
+    let expected = Some(10988901);
+    assert_eq!(output, expected);
 }
```

Comment:
This test case is new, should I still add the "no important files changed" flag?

Reply:
Ehm, probably not, as you want to re-run the test runner for those solutions.